src/query.ts

```diff
@@ -45,7 +45,7 @@ export function parseQuery<T extends ParsedQuery = ParsedQuery>(
       continue;
     }
     const value = decodeQueryValue(s[2] || "");
-    if (object[key] === undefined) {
+    if (!Object.prototype.hasOwnProperty.call(object, key)) {
```

Comment:
We might want to use [`Object.hasOwn()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) static method, which is intended as a replacement for [Object.prototype.hasOwnProperty()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) in this kind of case.

Reply:
I've considered `Object.hasOwn()`, but its [compatibility](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn#browser_compatibility) isn't that good. 

If its acceptable for this lib, then `Object.hasOwn()` would be a better choice ❤️
WDYT?

<img width="843" alt="image" src="https://github.com/user-attachments/assets/8294bb8c-26ae-4fc9-8d0f-2395f6faba12" />

Reply:
> We need to use an object with null proto.

Done.